### PR TITLE
Add containerization extras tests

### PR DESCRIPTION
**Pull request by KushagraSikka**
please verify,  it requires Swift 6.2 to run, which aligns with the project's stated requirements.

1. 📋 Test Coverage Created

- TestAsyncLock.swift (13 comprehensive tests):
- Basic locking functionality
- Sequential vs concurrent access patterns
- Error handling and recovery
- FIFO ordering and reentrancy prevention
- Performance characteristics
- Multiple lock instances
- Complex return types and edge cases

2. TestTimeout.swift (15 comprehensive tests):

-  Successful operations and quick operations
-  Timeout behavior and accuracy
-  Error propagation and concurrent timeouts
-  Task cancellation and cleanup
-  Performance and resource management

 Edge cases (zero timeout, large timeout, consecutive timeouts)

### CODE REVIEW

**Comment by dcantah** [comment]
This is awesome! I'll take a look at this sometime between today and tomorrow. Thanks a ton!

File: Tests/ContainerizationExtrasTests/TestAsyncLock.swift [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+        }
+        
+        let tracker = ExecutionTracker()
+        
+        await withTaskGroup(of: Void.self) { group in
+            // First task - holds lock for a while
+            group.addTask {
+                await lock.withLock { context in
+                    await tracker.append("task1-start")
+                    try? await Task.sleep(nanoseconds: 50_000_000) // 50ms
```
Comment (dcantah) [inline Tests/ContainerizationExtrasTests/TestAsyncLock.swift:234]: Task.sleep has a for: constructor that takes in enums for different time intervals and should look much nicer:

`try await Task.sleep(for: .milliseconds(50))`

Comment (dcantah) [reply Tests/ContainerizationExtrasTests/TestAsyncLock.swift:234]: Same advice for the other sleeps

**Comment by dcantah** [comment]
@dkovba Wanna look at the AsyncLock documentation updates here?

File: Sources/ContainerizationExtras/AsyncLock.swift [hunk 2]
```diff
@@ -16,21 +16,95 @@
...
+/// ```swift
+/// actor ResourceManager {
+///     private let lock = AsyncLock()
+///     private var resources: [String] = []
+///
+///     func addResource(_ name: String) async {
+///         await lock.withLock { context in
+///             // Async operations are safe within the lock
+///             let processedName = await processResourceName(name)
+///             resources.append(processedName)
```
Comment (dkovba) [inline Sources/ContainerizationExtras/AsyncLock.swift:41]: Should we add `func appendResource(_ resource: String, context: AsyncLock.Context)` to demonstrate how `context` is used? The required context parameter ensures that the function is called only inside a lock.

File: Sources/ContainerizationExtras/AsyncLock.swift [hunk 3]
```diff
@@ -16,21 +16,95 @@
...
+/// mutual exclusion without blocking threads. Operations are queued and resumed
+/// sequentially as the lock becomes available.
 public actor AsyncLock {
     private var busy = false
     private var queue: ArraySlice<CheckedContinuation<(), Never>> = []
 
+    /// A context object provided to closures executed within the lock.
+    ///
+    /// The context serves as proof that the code is executing within the lock's
+    /// critical section. While currently empty, it may be extended in the future
```
Comment (dkovba) [inline Sources/ContainerizationExtras/AsyncLock.swift:65]: It is empty by design because this is sufficient to serve as proof that the code is executed within the lock.

**Review by dkovba — COMMENTED** [review]
👍

**Comment by dkovba** [comment]
Please run `make fmt`.

**Comment by crosbymichael** [comment]
@KushagraSikka Thank you for the pull request.  Do you have time to update this with the provided feedback?

**Comment by egernst** [comment]
@KushagraSikka - adding more tests is a great contribution and I appreciate the first effort. Let us know if you're able to address the feedback, otherwise we may need someone else to help take this on.

**Comment by dcantah** [comment]
Going to close this out, if we revisit something similar we'll make sure to try to give you credit.
